mobile/mobile.go

```diff
@@ -50,6 +50,8 @@ type mobileClient struct {
 	localPrivCreateCallback  NativeCallback
 	remoteKeyReceiveCallback NativeCallback
 	authDataCallback         NativeCallback
+
+	mutex sync.Mutex
```

Comment:
nit: since only users of `mobileClient` use the mutex, you can just embed it into the struct. That way callers can just do `mc.Lock()` instead of `mc.mutex.Unlock()`